platforms/cortex/platform.c

```diff
@@ -263,9 +232,67 @@ static void *_task_planner_start_cancel(void *arg)
     return NULL;
 }
 
-void pf_init_tasks(void)
+void pf_init(void)
 {
+#ifdef MODULE_SHELL_PLATFORMS
+    pf_shell_init();
+#endif /* MODULE_SHELL_PLATFORMS */
+
+    /* Debug LED */
+    if (gpio_init(GPIO_DEBUG_LED, GPIO_OUT)) {
+        puts("WARNING: GPIO_DEBUG_LED not initialized !");
+    }
+    gpio_clear(GPIO_DEBUG_LED);
+
+    motor_driver_init(MOTOR_DRIVER_DEV(0));
+
+    /* Setup qdec periphereal */
+    int error = qdec_init(QDEC_DEV(HBRIDGE_MOTOR_LEFT), QDEC_MODE, NULL, NULL);
+    if (error) {
+        printf("QDEC %u not initialized, error=%d !!!\n", HBRIDGE_MOTOR_LEFT, error);
+    }
+    error = qdec_init(QDEC_DEV(HBRIDGE_MOTOR_RIGHT), QDEC_MODE, NULL, NULL);
+    if (error) {
+        printf("QDEC %u not initialized, error=%d !!!\n", HBRIDGE_MOTOR_RIGHT, error);
+    }
+
+    /* Init odometry */
+    odometry_setup(WHEELS_DISTANCE / PULSE_PER_MM);
+
+    /* Init starter and camp selection GPIOs */
+    if (gpio_init(GPIO_CAMP, GPIO_IN) != 0) {
+        puts("WARNING: GPIO_CAMP not initialized !");
+    }
+    if (gpio_init(GPIO_STARTER, GPIO_IN_PU) != 0) {
+        puts("WARNING: GPIO_STARTER not initialized !");
+    }
 
+
+    gpio_clear(GPIO_DEBUG_LED);
+
+    ctrl_set_anti_blocking_on(pf_get_ctrl(), TRUE);
+
+    /* Get platform path */
+    path_t *path = pf_get_path();
+    if (!path) {
+        printf("machine has no path\n");
+    }
```

Comment:
Should add an assert() to check path object here. The path is loaded right after on line 283 and can lead to a pointer exception.

Maybe previous gpio_init() checks can be done with assert() too but I don't know if it's a good idea or not